test/modules/MetricsValidator/src/Program.cs

```diff
@@ -71,10 +71,16 @@ static async Task<int> MainAsync()
                                 await Task.WhenAll(tests.Select(test => test.Start(cts.Token)));
                             }
 
-                            return new MethodResponse(Encoding.UTF8.GetBytes(testReporter.ReportResults()), (int)HttpStatusCode.OK);
+                            var result = new MethodResponse(Encoding.UTF8.GetBytes(testReporter.ReportResults()), (int)HttpStatusCode.OK);
+
+                            Logger.LogInformation($"Finished validating metrics. Result size: {result.Result.Length}");
+                            return result;
                         },
                         null);
 
+                    moduleClient.SetConnectionStatusChangesHandler((status, reason)
+                        => Logger.LogWarning($"IoT Hub connection status Changed Status: {status} Reason: {reason}"));
```

Comment:
Won't this detect connection status changed with EH? And if Hub closes the connection doesn't EH try to recreate it without disconnecting the module and triggering this callback?

Reply:
Yes, that's true. But this is for unlikely case when the module itself looses the connection.

Reply:
I was trying to say that the warning is indicating a connection status with IoTHub, but this callback is for connection status changes with module->EH. Should we change the warning name?

Reply:
Ah, I see. I've updated the message.